### PR TITLE
Bump up Clarity to the latest version

### PR DESCRIPTION
**Pull request by antgamdia**
_Depends on #2442_ 

### Description of the change

This PR bumps the clarity versions up, changes every clr->cds import, add some css imports and minor fixes.

### Benefits

This PR paves the way for the upgrading process from clarity v4 to v5. 

### Possible drawbacks

This PR (and some of the subsequent ones) are aimed at allowing dark mode in clarity v5. Until every PR becomes merged, some css inconsistencies might occur.

### Applicable issues

  - related #2372

### Additional information

The whole set of PRs includes:

- Remove hardcoded css, delegate colors in the cds variables (which is more customizable through themes), minor css fixes.
- Bump up dependencies. Minor fixes.
  - Update modals
  - Update layout
  - Add media query to detect dark mode (for cds)
  - Add dynamic dark mode for clr

### CODE REVIEW

File: dashboard/src/components/Layout/Clarity.tsx [hunk 1]
```diff
@@ -29,11 +29,11 @@ import {
...
   trashIcon,
   uploadCloudIcon,
-} from "@clr/core/icon";
-import "@clr/core/icon/register.js";
+} from "@cds/core/icon";
+import "@cds/core/icon/register.js";
 import * as React from "react";
 
-import "@clr/ui/clr-ui.min.css";
+import "@clr/ui/clr-ui.min.css"; // light clarity UI theme
```
Comment (andresmgot) [inline dashboard/src/components/Layout/Clarity.tsx:36]: shouldn't we use just one place to import clarity css files? (either here or in the index file, that btw is failing to import the files).

**Review by andresmgot — APPROVED** [review]
what's the difference between `@clr` and `@cds` packages? (they seem like mirrors but I guess they are different).

File: dashboard/src/components/Layout/Clarity.tsx (hunk 1 above)
Comment (antgamdia) [reply dashboard/src/components/Layout/Clarity.tsx:36]: Yep, this is exactly what we are doing in https://github.com/kubeapps/kubeapps/pull/2450
I just left unchanged this file (apart from clarifying it is the light theme)

**Comment by antgamdia** [comment]
> what's the difference between `@clr` and `@cds` packages? (they seem like mirrors but I guess they are different).

From the [website](https://github.com/vmware/clarity/tree/next): 

`@clr/icons` Contains the custom element icons.
`@clr/ui` Contains the static styles for building HTML components.
`@clr/angular` Contains the Angular components. This package depends on @clr/ui for styles.
`@cds/core` Contains the Web Components that work in any JavaScript framework.

There is also `@cds/react`, which is the official wrapper of these components.

So, we are using all except for the angular one.

File: dashboard/src/components/AppView/AccessURLTable/__snapshots__/AccessURLTable.test.tsx.snap [hunk 2]
```diff
@@ -371,7 +371,20 @@ exports[`when the app contains ingresses should show the table with available in
                                 size="sm"
                                 solid={true}
                               >
-                                <cds-icon />
```
Comment (andresmgot) [inline dashboard/src/components/AppView/AccessURLTable/__snapshots__/AccessURLTable.test.tsx.snap:374]: can you delete the screenshot instead :) ?

**Comment by antgamdia** [comment]
FYI @andresmgot , I had to replace the cds button (the one for adding docker credentials when adding a new repo) by a normal button after hitting the same issue you face in the past. The form was being prematurely submitted. I guess we should replace the current `clr`form with the entire `cds` one, but dunno :s (out of the scope of this PR though)

**Comment by andresmgot** [comment]
> FYI @andresmgot , I had to replace the cds button (the one for adding docker credentials when adding a new repo) by a normal button after hitting the same issue you face in the past. The form was being prematurely submitted. I guess we should replace the current `clr`form with the entire `cds` one, but dunno :s (out of the scope of this PR though)

Weird, maybe the `@clr` package had the fix but the `cds` one not? Anyway, I guess this is fine until we investigate the issue

**Pull request merged**